### PR TITLE
feat: updated model card for qwen_2.5_vl

### PR DESCRIPTION
**Pull request by arkhamHack**
# What does this PR do?

https://github.com/huggingface/transformers/issues/36979
This pr updates the model card for Qwen2.5-VL

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: docs/source/en/model_doc/qwen2_5_vl.md [hunk 1]
```diff
@@ -14,166 +14,109 @@ rendered properly in your Markdown viewer.
...
+<img alt="SDPA" src="https://img.shields.io/badge/SDPA-DE3412?style=flat&logo=pytorch&logoColor=white">    </div>
 </div>
 
-## Overview
-
-The [Qwen2.5-VL](https://qwenlm.github.io/blog/qwen2_5-vl/) model is an update to [Qwen2-VL](https://arxiv.org/abs/2409.12191) from Qwen team, Alibaba Group. 
+# Qwen2.5-VL
 
-The abstract from this update is the following:
+The [Qwen2.5-VL](https://qwenlm.github.io/blog/qwen2_5-vl/) model is a multimodal vision-language model developed by the Qwen team, Alibaba Group, combining an enhanced ViT encoder with the Qwen2.5 LLM. As an update to [Qwen2-VL](https://arxiv.org/abs/2409.12191), it offers improved visual reasoning and video understanding capabilities. The model uses a refined ViT architecture with SwiGLU and RMSNorm, making it more aligned with the LLM’s structure.
```
Comment (stevhliu) [inline docs/source/en/model_doc/qwen2_5_vl.md:26]:
```suggestion
[Qwen2.5-VL](https://huggingface.co/papers/2502.13923) is a multimodal vision-language model, available in 3B, 7B, and 72B parameters, pretrained on 4.1T tokens. The model introduces window attention in the ViT encoder to accelerate training and inference, dynamic FPS sampling on the spatial and temporal dimensions for better video understanding across different sampling rates, and an upgraded MRoPE (multi-resolutional rotary positional encoding) mechanism to better capture and learn temporal dynamics.
```

File: docs/source/en/model_doc/qwen2_5_vl.md [hunk 2]
```diff
@@ -14,166 +14,109 @@ rendered properly in your Markdown viewer.
...
 
-*Qwen2.5-VL marks a major step forward from Qwen2-VL, built upon the latest Qwen2.5 LLM. We've accelerated training and testing through the strategic implementation of window attention within the ViT. The ViT architecture itself has been refined with SwiGLU and RMSNorm, aligning it more closely with the LLM's structure. A key innovation is the expansion of native dynamic resolution to encompass the temporal dimension, in addition to spatial aspects. Furthermore, we've upgraded MRoPE, incorporating absolute time alignment on the time axis to allow the model to effectively capture temporal dynamics, regardless of frame rate, leading to superior video understanding.*
+Qwen2.5-VL introduces window attention in the ViT, accelerating both training and inference. It also supports dynamic resolution across both spatial and temporal dimensions, making it highly effective for video analysis. The upgraded MRoPE (Multi-Resolutional Rotary Positional Encoding) now includes absolute time alignment on the time axis, enabling it to capture temporal dynamics across varying frame rates, enhancing its video comprehension abilities.
```
Comment (stevhliu) [inline docs/source/en/model_doc/qwen2_5_vl.md:28]:
```suggestion
```

File: docs/source/en/model_doc/qwen2_5_vl.md [hunk 3]
```diff
@@ -14,166 +14,109 @@ rendered properly in your Markdown viewer.
...
-output_ids = model.generate(**inputs, max_new_tokens=128)
-generated_ids = [output_ids[len(input_ids):] for input_ids, output_ids in zip(inputs.input_ids, output_ids)]
-output_text = processor.batch_decode(generated_ids, skip_special_tokens=True, clean_up_tokenization_spaces=True)
+output_text = processor.batch_decode(
+       generated_ids_trimmed, skip_special_tokens=True, clean_up_tokenization_spaces=False
+)
 print(output_text)
 ```
+</hfoption>
+</hfoptions>
```
Comment (stevhliu) [inline docs/source/en/model_doc/qwen2_5_vl.md:115]: Let's also include a quantization example for text generation with the 7B model

File: docs/source/en/model_doc/qwen2_5_vl.md [hunk 4]
```diff
@@ -14,166 +14,109 @@ rendered properly in your Markdown viewer.
...
+
+### Notes
 
-### Usage Tips
+Qwen2.5-VL is a multimodal conversational model designed for image, video, and text understanding, making it highly versatile for both general-purpose and fine-tuned downstream tasks such as image captioning, visual question answering (VQA), scene description, and video understanding.
```
Comment (stevhliu) [inline docs/source/en/model_doc/qwen2_5_vl.md:119]: Add the relevant (indented) code examples under each of these points.

```suggestion
- Use Qwen2.5-VL for video inputs by setting `"type": "video"` as shown below.
- Use Qwen2.5-VL for a mixed batch of inputs (images, videos, text) as show below.
- Use the `min_pixels` and `max_pixels` parameters in [`AutoProcessor`] to set the resolution. Higher resolution can require more compute whereas reducing the resolution can save memory.
- Add labels when handling multiple images or videos for better reference as shown below.
```

**Review by stevhliu — COMMENTED** [review]
Super nice, just missing a few notes and quantization example!

**Comment by arkhamHack** [comment]
Hi @stevhliu,  I made the changes you requested. Could you check now? 
Thanks.

**Comment by arkhamHack** [comment]
hi @stevhliu . I have made the changes you requested. Could you review this pr? Thanks.

File: docs/source/en/model_doc/qwen2_5_vl.md [hunk 5]
```diff
@@ -14,255 +14,233 @@ rendered properly in your Markdown viewer.
...
-    {
-        "role": "user",
-        "content": [
-            {"type": "video", "path": "/path/to/video.mp4"},
-            {"type": "text", "text": "What happened in the video?"},
-        ],
-    }
 ]
 
+image = Image.open(requests.get("https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/pipeline-cat-chonk.jpeg", stream=True).raw)
```
Comment (stevhliu) [inline docs/source/en/model_doc/qwen2_5_vl.md:95]: Don't need this

```suggestion
```

File: docs/source/en/model_doc/qwen2_5_vl.md [hunk 6]
```diff
@@ -14,255 +14,233 @@ rendered properly in your Markdown viewer.
...
-conversation1 = [
-    {
-        "role": "user",
-        "content": [
-            {"type": "image", "path": "/path/to/image1.jpg"},
-            {"type": "text", "text": "Describe this image."}
-        ]
-    }
-]
+).to("cuda:0")
```
Comment (stevhliu) [inline docs/source/en/model_doc/qwen2_5_vl.md:102]:
```suggestion
).to("cuda")
```

File: docs/source/en/model_doc/qwen2_5_vl.md [hunk 7]
```diff
@@ -14,255 +14,233 @@ rendered properly in your Markdown viewer.
...
-
-```bash
-pip install -U flash-attn --no-build-isolation
-```
-
-Also, you should have hardware that is compatible with FlashAttention 2. Read more about it in the official documentation of the [flash attention repository](https://github.com/Dao-AILab/flash-attention). FlashAttention-2 can only be used when a model is loaded in `torch.float16` or `torch.bfloat16`.
-
-To load and run a model using FlashAttention-2, add `attn_implementation="flash_attention_2"` when loading the model:
+Quantization reduces the memory burden of large models by representing the weights in a lower precision. Refer to the [Quantization](../quantization/overview) overview for more available quantization backends.
+The example below uses [torchao](../quantization/torchao) to only quantize the weights to int4.
```
Comment (stevhliu) [inline docs/source/en/model_doc/qwen2_5_vl.md:117]:
```suggestion
Quantization reduces the memory burden of large models by representing the weights in a lower precision. Refer to the [Quantization](../quantization/overview) overview for more available quantization backends.

The example below uses [torchao](../quantization/torchao) to only quantize the weights to int4.
```

File: docs/source/en/model_doc/qwen2_5_vl.md [hunk 8]
```diff
@@ -14,255 +14,233 @@ rendered properly in your Markdown viewer.
...
 
 ```python
-from transformers import Qwen2_5_VLForConditionalGeneration
-
+quantization_config = TorchAoConfig("int4_weight_only", group_size=128)
```
Comment (stevhliu) [inline docs/source/en/model_doc/qwen2_5_vl.md:120]: Complete this code snippet so users can run it end to end

File: docs/source/en/model_doc/qwen2_5_vl.md [hunk 9]
```diff
@@ -14,255 +14,233 @@ rendered properly in your Markdown viewer.
...
+    ```
+    
+    Higher resolution can require more compute whereas reducing the resolution can save memory as follows:
+    
+    ```python
+    min_pixels = 256*28*28
+    max_pixels = 1024*28*28 
+    processor = AutoProcessor.from_pretrained("Qwen/Qwen2.5-VL-7B-Instruct", min_pixels=min_pixels, max_pixels=max_pixels)
+    ```
+- We can implement flash attention 2 to speed up generation as shown below.
```
Comment (stevhliu) [inline docs/source/en/model_doc/qwen2_5_vl.md:228]: I think we can remove this note

File: docs/source/en/model_doc/qwen2_5_vl.md [hunk 10]
```diff
@@ -14,255 +14,233 @@ rendered properly in your Markdown viewer.
...
+<hfoption id="Pipeline">
 
+```py
 import torch
-from transformers import Qwen2_5_VLForConditionalGeneration, AutoTokenizer, AutoProcessor
+from transformers import pipeline
+pipe = pipeline(
+    task="image-text-to-text",
+    model="Qwen/Qwen2.5-VL-7B-Instruct",
+    device="cuda",
```
Comment (stevhliu) [inline docs/source/en/model_doc/qwen2_5_vl.md:45]:
```suggestion
    device=0,
```

**Review by stevhliu — COMMENTED** [review]
Thanks, just a few more changes to go!

File: docs/source/en/model_doc/qwen2_5_vl.md (hunk 9 above)
Comment (arkhamHack) [reply docs/source/en/model_doc/qwen2_5_vl.md]: ive removed the flash attention 2 note from here . hope the code snippet before that is ok to stay.

**Comment by arkhamHack** [comment]
Hi @stevhliu made the remaining changes.

**Review by stevhliu — APPROVED** [review]
LGTM, thanks for your contribution!
